opty/direct_collocation.py

```diff
@@ -188,7 +203,30 @@ def __init__(self, equations_of_motion, state_symbols,
             to a directory here.
 
         """
-        self.eom = equations_of_motion
+        Lambda_Matrix = sym.Matrix([[0,0,0,0], [0,0,0,0], [0,0,Lambda,0],[0,0,0,Lambda]])
+	    """
+	    define matrix Lambda =   [0 0 0      0     ]
+			             [0 0 0      0     ]
+ 			             [0 0 lambda 0     ]
+			             [0 0 0      lambda]
+	    """
+        K_Matrix = sym.Matrix([[0,0,0,0], [0,0,0,0], [0,0,K,0],[0,0,0,K]])
+	    """
+	    define matrix Lambda =   [0 0 0 0 ]
+			             [0 0 0 0 ]
+ 			             [0 0 K 0 ]
+			             [0 0 0 K ]
+	    """
```

Comment:
Yes, your way is more general.

On Thu, May 19, 2016 at 2:41 PM, Jason K. Moore <notifications@github.com>
wrote:

> In opty/direct_collocation.py
> <https://github.com/csu-hmc/opty/pull/40#discussion_r63932064>:
>
> > @@ -188,7 +203,30 @@ def __init__(self, equations_of_motion, state_symbols,
> >              to a directory here.
> >
> >          """
> > -        self.eom = equations_of_motion
> > +        Lambda_Matrix = sym.Matrix([[0,0,0,0], [0,0,0,0], [0,0,Lambda,0],[0,0,0,Lambda]])
> > +	    """
> > +	    define matrix Lambda =   [0 0 0      0     ]
> > +			             [0 0 0      0     ]
> > + 			             [0 0 lambda 0     ]
> > +			             [0 0 0      lambda]
> > +	    """
> > +        K_Matrix = sym.Matrix([[0,0,0,0], [0,0,0,0], [0,0,K,0],[0,0,0,K]])
> > +	    """
> > +	    define matrix Lambda =   [0 0 0 0 ]
> > +			             [0 0 0 0 ]
> > + 			             [0 0 K 0 ]
> > +			             [0 0 0 K ]
> > +	    """
>
> I think these need to be something like:
>
> k_matrix = sym.zeros(2 * num_states)
> k_matrix[num_states:, num_states:] = k * sym.eye(num_states)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/csu-hmc/opty/pull/40/files/8f115d75b841d3daf82afb0069ffdd766dba99ef#r63932064>
>